### DISCUSSION TITLE
Any migration after init always fails to apply cleanly

### DISCUSSION THREAD

**Discussion by Afulton11** [prisma/prisma#22731] [Q&A]
### Question

_Why is Prisma unable to create a new migration in this case?_

#### Other Info
1. Running `prisma db push` succeeds as `The database is already in sync with the Prisma schema.`
2. I've tried running `prisma migrate resolve --applied 0_init` and the init migration is still re-applied when running `prisma migrate dev`

### How to reproduce (optional)

1. Create Prisma Schema
3. Perform `prisma migrate dev --name init` to create the initial migration
4. Attempt to create a new migration, either by changing the schema and re-running or by using `--create-only` option.
    * i.e: `prisma migrate dev --name adds_something --create-only`
5. Prisma throws error like:
```
Migration `123_init` failed to apply cleanly to the shadow database.
ERROR: type "Abc" already exists
    0: schema_core::state::DevDiagnostic
             at schema-engine/core/src/state.rs:267
```

### Expected behavior (optional)

Prisma skips the initial migration as it is already applied and creates a new migration.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["views", "fullTextSearch", "nativeDistinct", "relationJoins", "postgresqlExtensions"]
}

datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
  extensions = [pg_trgm, uuid_ossp(map: "uuid-ossp", schema: "extensions"), pg_graphql, pg_stat_statements, pgcrypto, pgjwt, pgsodium, plpgsql, supabase_vault]
}

enum Abc {
  A
  B
  C
}

// ... other models
```

#### OS
`macOS 12.7.1`

#### Database
Supabase PostgreSQL

#### Node Version
`v20.9.0`

#### Prisma Version
```
prisma                  : 5.8.1
@prisma/client          : 5.8.1
Computed binaryTarget   : darwin
Operating System        : darwin
Architecture            : x64
Node.js                 : v20.9.0
Query Engine (Node-API) : libquery-engine 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at ../../node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli 78caf6feeaed953168c64e15a249c3e9a033ebe2 (at ../../node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.8.1-1.78caf6feeaed953168c64e15a249c3e9a033ebe2
Default Engines Hash    : 78caf6feeaed953168c64e15a249c3e9a033ebe2
Studio                  : 0.497.0
Preview Features        : fullTextSearch, nativeDistinct, postgresqlExtensions, views, relationJoins
```

**Comment by janpio** [comment]
> Migration `123_init` failed to apply cleanly to the shadow database.

The important term here is "shadow database". `migrate dev` always applies all migrations from the migrations folder to the _shadow database_ to get to the state that your database is _supposed_ to be in. Then it compares that to the real state, and throws a drift error if there is a different to prevent you from applying a migration while your database is in an unclean state.

The shadow database is an empty database usually. You'll have to figure out why this type `Abc` already exists when it tried to apply the `init` migrations, that seems to recreate it again.

**Reply by Susanta-0007** [reply]
i am also have error

**Comment by Susanta-0007** [comment]
![Screenshot 2024-01-28 225232](https://github.com/prisma/prisma/assets/89796600/19c7ff8f-b29a-41f0-bb40-81707deb0717)

**Reply by Afulton11** [reply]
I ended up solving the issue by using a separate postgresql database for my shadow database.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
